sys/net/application_layer/dhcpv6/client.c

```diff
@@ -247,6 +247,17 @@ static inline size_t _compose_elapsed_time_opt(dhcpv6_opt_elapsed_time_t *time)
     return len + sizeof(dhcpv6_opt_t);
 }
 
+static inline size_t _compose_mud_url_opt(dhcpv6_opt_mud_url_t *mud_url_opt,
+                                          char mud_url[])
+{
+    uint16_t len = strlen(mud_url);
+
+    mud_url_opt->type = byteorder_htons(DHCPV6_OPT_MUD_URL);
+    mud_url_opt->len = byteorder_htons(len);
+    strcpy(mud_url_opt->mudString, mud_url);
```

Comment:
no length check if there is enough space left in the destination buffer?

Reply:
Oh, good point. Would using `strncpy` instead `strcpy` solve this problem?

Reply:
I also noticed that the current buffer length is only 256 Bytes long.

https://github.com/RIOT-OS/RIOT/blob/cfdf30409a664a3d498b68eb4d2ea8697e0bc13c/sys/include/net/dhcpv6/client.h#L48

I assume the length of the send buffer should be increased by 255 Bytes if the `gnrc_dhcpv6_client_mud_url` is used?

Reply:
> Would using `strncpy` instead of `strcpy` solve this problem?

Only if truncated URLs are still of any value

> I assume the length of the send buffer should be increased by 255 Bytes (or in accordance to the total length of the MUD option) if the `gnrc_dhcpv6_client_mud_url` pseudo-module is used?

Yea something like

```
#ifdef MODULE_GNRC_DHCPV6_CLIENT_MUD_URL
#define DHCPV6_CLIENT_BUFLEN        (512)
#else 
#define DHCPV6_CLIENT_BUFLEN        (256)
#endif
```

although it would probably be enough to only enlarge the send buffer

Reply:
> 
> 
> > Would using `strncpy` instead of `strcpy` solve this problem?
> 
> Only if truncated URLs are still of any value

Hmm, probably not. The enlarged buffer should, however, minimize the risk of truncation. Or is there maybe a more robust and/or elegant way to prevent such a scenario?

> 
> > I assume the length of the send buffer should be increased by 255 Bytes (or in accordance to the total length of the MUD option) if the `gnrc_dhcpv6_client_mud_url` pseudo-module is used?
> 
> Yea something like
> 
> ```c
> #ifdef MODULE_GNRC_DHCPV6_CLIENT_MUD_URL
> #define DHCPV6_CLIENT_BUFLEN        (512)
> #else 
> #define DHCPV6_CLIENT_BUFLEN        (256)
> #endif
> ```
> 
> although it would probably be enough to only enlarge the send buffer

Would the addition of something like

```c
#ifdef MODULE_GNRC_DHCPV6_CLIENT_MUD_URL
#define DHCPV6_CLIENT_SEND_BUFLEN        (DHCPV6_CLIENT_BUFLEN)
#else 
#define DHCPV6_CLIENT_SEND_BUFLEN        (DHCPV6_CLIENT_BUFLEN + 256)
#endif
```

in `client.c` be a good solution?